src/containerapp/azext_containerapp/tests/latest/test_containerapp_connected_env_scenario.py

```diff
@@ -0,0 +1,369 @@
+
+        self.cmd('containerapp connected-env dapr-component list -n {} -g {}'.format(env_name, resource_group), checks=[
+            JMESPathCheck('length(@)', 0),
+        ])
+
+    @serial_test()
+    @live_only()  # generate_randomized_cert_name cause No match for the request (<Request (PUT) /my-connected-env/certificates/my-connected-e-clitest.rg0000-8d2d-6528?
+    @ResourceGroupPreparer(location="eastus", random_name_length=15)
+    @ConnectedClusterPreparer(location=TEST_LOCATION)
+    def test_containerapp_preview_connected_env_certificate(self, resource_group, connected_cluster_name):
+        self.cmd('configure --defaults location={}'.format(TEST_LOCATION))
+        custom_location_name = "my-custom-location"
+        create_extension_and_custom_location(self, resource_group, connected_cluster_name, custom_location_name)
+
+        # create connected environment with client or create a command for connected?
+        sub_id = self.cmd('az account show').get_output_in_json()['id']
+        env_name = 'my-connected-env'
+        custom_location_id = f"/subscriptions/{sub_id}/resourceGroups/{resource_group}/providers/Microsoft.ExtendedLocation/customLocations/{custom_location_name}"
+        self.cmd(
+            f'containerapp connected-env create -g {resource_group} --name {env_name} --custom-location {custom_location_name} -l {TEST_LOCATION}',
+            checks=[
+                JMESPathCheck('name', env_name),
+                JMESPathCheck('properties.provisioningState', "Succeeded"),
+                JMESPathCheck('extendedLocation.name', custom_location_id)
+            ])
+        self.cmd('containerapp connected-env certificate list -g {} -n {}'.format(resource_group, env_name), checks=[
+            JMESPathCheck('length(@)', 0),
+        ])
+
+        # test that non pfx or pem files are not supported
+        txt_file = os.path.join(TEST_DIR, 'cert.txt')
+        self.cmd('containerapp connected-env certificate upload -g {} -n {} --certificate-file "{}"'.format(resource_group, env_name, txt_file), expect_failure=True)
+
+        # test pfx file with password
+        pfx_file = os.path.join(TEST_DIR, 'cert.pfx')
+        pfx_password = 'test12'
+        cert = self.cmd('containerapp connected-env certificate upload -g {} -n {} --certificate-file "{}" --password {}'.format(
+            resource_group, env_name, pfx_file, pfx_password), checks=[
+            JMESPathCheck('properties.provisioningState', "Succeeded"),
+            JMESPathCheck('type', "Microsoft.App/connectedEnvironments/certificates"),
+        ]).get_output_in_json()
+
+        cert_name = cert["name"]
+        cert_id = cert["id"]
+        cert_thumbprint = cert["properties"]["thumbprint"]
+        cert_location = cert["location"]
+
+        self.cmd('containerapp connected-env certificate list -n {} -g {} -l "{}"'.format(env_name, resource_group, cert_location), checks=[
```

Comment:
-l

Reply:
This is by designed, there is some logic for list certificate with `location `